test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemJSTest.scala

```diff
@@ -57,4 +59,56 @@ class SystemJSTest {
       assertEquals(x1FirstHash, System.identityHashCode(x1))
     }
   }
+
+  @Test def systemProperties(): Unit = {
+    def get(key: String): String = java.lang.System.getProperty(key)
+
+    // Defined in System.scala
+    assertEquals("1.8", get("java.version"))
+    assertEquals("1.8", get("java.vm.specification.version"))
+    assertEquals("Oracle Corporation", get("java.vm.specification.vendor"))
+    assertEquals("Scala.js", get("java.vm.name"))
+    assertEquals("1.8", get("java.specification.version"))
+    assertEquals("/", get("file.separator"))
+    assertEquals(":", get("path.separator"))
+    assertEquals("\n", get("line.separator"))
+
+    // Added via __ScalaJSEnv
+
+    assertNotNull(get("java.vm.version"))
+    assertNotNull(get("os.name"))
+    assertNotNull(get("os.version"))
+    assertNotNull(get("os.arch"))
+    assertNotNull(get("scalajs.compliant-moduleinit"))
+    assertNotNull(get("scalajs.compliant-asinstanceofs"))
+    assertNotNull(get("scalajs.production-mode"))
+
+    val inNode = get("scalajs.node") == "true"
+    val inPhantom = get("scalajs.phantom") == "true"
+    val inRhino = get("scalajs.rhino") == "true"
+    val hasTypedArrays = get("scalajs.typedarray") == "true"
+
+    if (inNode) {
+      val process = js.Dynamic.global.process
+      assertFalse(js.isUndefined(process))
+      assertEquals("node", process.title)
```

Comment:
I am pretty sure this can be wrong. For example if Node.js is launched via a symbolic link `nodejs`. This test should be removed.